### PR TITLE
feat(components/forms): add help inline inputs to checkbox

### PR DESCRIPTION
**Pull request by Blackbaud-SandhyaRajasabeson**
[AB#2184988](https://dev.azure.com/blackbaud/Products/_boards/board/t/SKY%20UX%20Program/Stories/?workitem=2184988)

### CODE REVIEW

**Comment by blackbaud-sky-build-user** [comment]
[Storybook preview](https://blackbaud.github.io/skyux-pr-preview/2247/storybook/)

Component Storybooks:

- [action-bars-storybook](https://blackbaud.github.io/skyux-pr-preview/2247/storybooks/action-bars-storybook/)
- [ag-grid-storybook](https://blackbaud.github.io/skyux-pr-preview/2247/storybooks/ag-grid-storybook/)
- [angular-tree-component-storybook](https://blackbaud.github.io/skyux-pr-preview/2247/storybooks/angular-tree-component-storybook/)
- [avatar-storybook](https://blackbaud.github.io/skyux-pr-preview/2247/storybooks/avatar-storybook/)
- [colorpicker-storybook](https://blackbaud.github.io/skyux-pr-preview/2247/storybooks/colorpicker-storybook/)
- [data-manager-storybook](https://blackbaud.github.io/skyux-pr-preview/2247/storybooks/data-manager-storybook/)
- [datetime-storybook](https://blackbaud.github.io/skyux-pr-preview/2247/storybooks/datetime-storybook/)
- [errors-storybook](https://blackbaud.github.io/skyux-pr-preview/2247/storybooks/errors-storybook/)
- [forms-storybook](https://blackbaud.github.io/skyux-pr-preview/2247/storybooks/forms-storybook/)
- [inline-form-storybook](https://blackbaud.github.io/skyux-pr-preview/2247/storybooks/inline-form-storybook/)
- [layout-storybook](https://blackbaud.github.io/skyux-pr-preview/2247/storybooks/layout-storybook/)
- [lists-storybook](https://blackbaud.github.io/skyux-pr-preview/2247/storybooks/lists-storybook/)
- [lookup-storybook](https://blackbaud.github.io/skyux-pr-preview/2247/storybooks/lookup-storybook/)
- [pages-storybook](https://blackbaud.github.io/skyux-pr-preview/2247/storybooks/pages-storybook/)
- [phone-field-storybook](https://blackbaud.github.io/skyux-pr-preview/2247/storybooks/phone-field-storybook/)
- [split-view-storybook](https://blackbaud.github.io/skyux-pr-preview/2247/storybooks/split-view-storybook/)
- [tabs-storybook](https://blackbaud.github.io/skyux-pr-preview/2247/storybooks/tabs-storybook/)
- [text-editor-storybook](https://blackbaud.github.io/skyux-pr-preview/2247/storybooks/text-editor-storybook/)
- [theme-storybook](https://blackbaud.github.io/skyux-pr-preview/2247/storybooks/theme-storybook/)

Apps:

- [code-examples](https://blackbaud.github.io/skyux-pr-preview/2247/code-examples/)
- [dep-graph](https://blackbaud.github.io/skyux-pr-preview/2247/dep-graph/)
- [integration](https://blackbaud.github.io/skyux-pr-preview/2247/integration/)
- [playground](https://blackbaud.github.io/skyux-pr-preview/2247/playground/)

File: libs/components/forms/testing/src/checkbox/checkbox-harness.ts [hunk 1]
```diff
@@ -35,6 +37,16 @@ export class SkyCheckboxHarness extends SkyComponentHarness {
     throw Error('No form errors found.');
   }
 
+  async #getHelpInline(): Promise<SkyHelpInlineHarness> {
```
Comment (Blackbaud-SteveBrush) [inline libs/components/forms/testing/src/checkbox/checkbox-harness.ts]: Private methods should go below public methods.

File: libs/components/forms/testing/src/checkbox/checkbox-harness.spec.ts [hunk 2]
```diff
@@ -1,5 +1,6 @@
 import { TestbedHarnessEnvironment } from '@angular/cdk/testing/testbed';
 import { TestBed } from '@angular/core/testing';
+import { NoopAnimationsModule } from '@angular/platform-browser/animations';
```
Comment (Blackbaud-SteveBrush) [inline libs/components/forms/testing/src/checkbox/checkbox-harness.spec.ts]: Let's import the `NoopAnimationsModule` into the `CheckboxHarnessTestModule` so that it has everything it needs to be imported into another TestBed (if needed).

File: apps/e2e/forms-storybook-e2e/src/e2e/checkbox.component.cy.ts [hunk 3]
```diff
@@ -15,6 +15,11 @@ describe('forms-storybook - checkbox', () => {
           .should('be.visible')
           .get('#touched-required-checkbox')
           .dblclick();
+        cy.get('sky-help-inline')
+          .eq(1)
+          .should('exist')
+          .should('be.visible')
+          .click();
```
Comment (Blackbaud-SteveBrush) [inline apps/e2e/forms-storybook-e2e/src/e2e/checkbox.component.cy.ts]: The popover seems to be in an invalid location on the screenshot? 
Screenshot 2024-04-19 at 12 00 13 PM

File: apps/e2e/forms-storybook/src/app/checkbox/checkbox.component.html [hunk 4]
```diff
@@ -80,6 +80,34 @@
...
+  <sky-row class="sky-padding-even-md">
+    <sky-column>
+      <sky-checkbox
+        labelText="Required checkbox with easy mode help inline"
+        helpPopoverContent="popover Content"
+        [required]="true"
+      >
+      </sky-checkbox>
+    </sky-column>
+  </sky-row>
```
Comment (Blackbaud-SteveBrush) [inline apps/e2e/forms-storybook/src/app/checkbox/checkbox.component.html:115]: I think it would also be good to have a screenshot of a required checkbox with inline help (like you have) but also with an error message to make sure all the pieces work well together.

Like this one, but with inline help:
Screenshot 2024-04-19 at 12 01 45 PM

File: apps/e2e/forms-storybook-e2e/src/e2e/checkbox.component.cy.ts (hunk 3 above)
Comment (Blackbaud-SandhyaRajasabeson) [reply apps/e2e/forms-storybook-e2e/src/e2e/checkbox.component.cy.ts]: ugh ok so this seems to be being caused by the screenshot needing to scroll to take the screenshot. It's working when I delete a bunch of other checkboxes to make the screenshot smaller. I'll have to ping john about this next week

Comment (Blackbaud-SandhyaRajasabeson) [reply apps/e2e/forms-storybook-e2e/src/e2e/checkbox.component.cy.ts]: actually I think im going to remove the checkbox with open help inline, changes to that functionality should not affect checkbox visual test.

**Comment by Blackbaud-TrevorBurch** [comment]
I know we have had a couple instances get into the CHANGELOG. However, we have mostly tried to not publicly call any of this "easy mode". I would reword the commit title to something like

`feat(components/forms): add help inline inputs to checkbox`

**Review by Blackbaud-TrevorBurch — DISMISSED** [review]

File: libs/components/forms/src/lib/modules/checkbox/checkbox.component.spec.ts [hunk 5]
```diff
@@ -1244,31 +1281,41 @@ describe('Checkbox component', () => {
     it('should set span class based on checkbox type input', () => {
```
Comment (Blackbaud-TrevorBurch) [inline libs/components/forms/src/lib/modules/checkbox/checkbox.component.spec.ts]: I know this wasn't you.... but could we potentially clean up this description a bit. Now that there are multiple `span` elements - it isn't very descriptive. Could we reword this to something like `should set the switch control's class based on checkbox type input`?

Comment (Blackbaud-SandhyaRajasabeson) [reply libs/components/forms/src/lib/modules/checkbox/checkbox.component.spec.ts]: good catch!

**Review by Blackbaud-TrevorBurch — APPROVED** [review]
